pkg/manifest/local_server_test.go

```diff
@@ -0,0 +1,214 @@
+package manifest
+
+import (
+	"bytes"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/pelletier/go-toml"
+)
+
+type KVWrapper struct {
+	KVStores LocalKVStoreMap `toml:"kv_stores"`
+}
+
+func (w KVWrapper) MarshalTOML() ([]byte, error) {
+	obj := make(map[string]any)
+	kv := make(map[string]any)
+
+	for key, entry := range w.KVStores {
+		if entry.External != nil {
+			kv[key] = map[string]any{
+				"file":   entry.External.File,
+				"format": entry.External.Format,
+			}
+		} else {
+			kv[key] = entry.Array
+		}
+	}
+
+	obj["kv_stores"] = kv
+
+	buf := new(bytes.Buffer)
+	err := toml.NewEncoder(buf).Encode(obj)
+	return buf.Bytes(), err
+}
+
+type SecretStoreWrapper struct {
+	SecretStores LocalSecretStoreMap `toml:"secret_stores"`
+}
+
+func (w SecretStoreWrapper) MarshalTOML() ([]byte, error) {
+	obj := make(map[string]any)
+	kv := make(map[string]any)
+
+	for key, entry := range w.SecretStores {
+		if entry.External != nil {
+			kv[key] = map[string]any{
+				"file":   entry.External.File,
+				"format": entry.External.Format,
+			}
+		} else {
+			kv[key] = entry.Array
+		}
+	}
+
+	obj["kv_stores"] = kv
+
+	buf := new(bytes.Buffer)
+	err := toml.NewEncoder(buf).Encode(obj)
+	return buf.Bytes(), err
+}
+
+func TestLocalKVStores_UnmarshalTOML(t *testing.T) {
+	tests := []struct {
+		name        string
+		inputTOML   string
+		expectError bool
+		expected    LocalKVStore
+	}{
+		{
+			name: "legacy array format",
+			inputTOML: `
+[[kv_stores.my-kv]]
+key = "kv"
+file = "kv.json"
+`,
+			expected: LocalKVStore{
+				IsArray: true,
+				Array: []KVStoreArrayEntry{
+					{
+						Key:  "kv",
+						File: "kv.json",
+					},
+				},
+			},
+		},
+		{
+			name: "external file format",
+			inputTOML: `
+[kv_stores]
+my-kv = { file = "kv.json", format = "json" }
+`,
+			expected: LocalKVStore{
+				IsArray: false,
+				External: &KVStoreExternalFile{
+					File:   "kv.json",
+					Format: "json",
+				},
+			},
+		},
+		{
+			name: "invalid format",
+			inputTOML: `
+[kv_stores]
+my-kv = "not-a-valid-entry"
+`,
+			expectError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var m KVWrapper
```

Comment:
This can be replaced with `LocalServer`.

Reply:
same

Reply:
I used an anonymous struct that looks like this:
```go
			var m struct {
				KVStores LocalKVStoreMap `toml:"kv_stores"`
			}
```

Since this test is supposed to test LocalKVStoreMap UnmarshalTOML only, this struct is beneficial for having just that layer.